routes/dashboard.rb

```diff
@@ -1,15 +1,34 @@
 class App < Sinatra::Base 
   get '/' do
     @cities = City.all
-    
-    @routes = Route.all.map do |route|
+
+    slim :index
+  end
+
+  post '/' do
+    if params[:departure_id].nil? || params[:departure_id].empty? || params[:arrival_id].nil? || params[:arrival_id].empty?
+      redirect '/'
+    else
+      query = request.params.map{|key, value| "#{key}=#{value}"}.join('&')
```

Comment:
`map{ => map {`
`{|key, value| => { |key, value| }`

Reply:
Extract to ~~service~~ Operation Object, looks weird inside controller